### PR TITLE
Integration of STL into EvalML

### PR DESCRIPTION
**Pull request by chukarsten**
Rebasing of Beccas #3781

### CODE REVIEW

File: evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def _project_trend(self, y):
                 )
                 - 1
             )
-        elif isinstance(y.index, Int64Index):
+        elif isinstance(y.index, Int64Index) or isinstance(y.index, RangeIndex):
```
Comment (chukarsten) [inline evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py:86]:
```suggestion
        elif isinstance(y.index, (RangeIndex, Int64Index)):
```

File: evalml/tests/pipeline_tests/test_pipeline_utils.py [hunk 2]
```diff
@@ -185,6 +188,59 @@ def test_make_pipeline(
...
+    }
+    estimators = get_estimators(problem_type=problem_type)
+    pipeline_class = _get_pipeline_base_class(problem_type)
+    for estimator_class in estimators:
+        if problem_type in estimator_class.supported_problem_types:
+
+            pipeline = make_pipeline(X, y, estimator_class, problem_type, parameters)
+            assert isinstance(pipeline, pipeline_class)
+
+            if is_regression(problem_type) and frequency in ["D", "MS"]:
```
Comment (chukarsten) [inline evalml/tests/pipeline_tests/test_pipeline_utils.py:227]: I thought we were catching "A" and "MS"?  If that's the case, I would think we might want to capture the list of frequencies as a module level list in the STLDecomposer (imported) and reference it here?  We can file that as a separate issue though.

Comment (eccabay) [reply evalml/tests/pipeline_tests/test_pipeline_utils.py:227]: Annual and ten minute intervals are the frequencies we've found to fail at this point. MS seems to fail with the polynomial decomposer with the millisecond issue but does fine with STL. I do have the failing frequencies in a constant within utils, I'll reference that here!

**Review by eccabay — APPROVED** [review]
Feels silly to approve a PR for my own work but here we are

**Pull request merged**